File: ce/snippets/visualbasic/CRMV8/queryvb/vb/linqexamples51.vb
' <snippetlinqexamples51>


 Using svcContext As New ServiceContext(_serviceProxy)
  Dim query_string = From c In svcContext.ContactSet _
                     Where c.ContactId.Value.Equals(_contactId2) _
                     Select New With
                            {Key .IndexOf = c.FirstName.IndexOf("contact"),
                             Key .Insert = c.FirstName.Insert(1, "Insert"),
                             Key .Remove = c.FirstName.Remove(1, 1),
                             Key .Substring = c.FirstName.Substring(1, 1),
                             Key .ToUpper = c.FirstName.ToUpper(),
                             Key .ToLower = c.FirstName.ToLower(),
                             Key .TrimStart = c.FirstName.TrimStart(),
                             Key .TrimEnd = c.FirstName.TrimEnd()}

  For Each c In query_string
   Console.WriteLine(c.IndexOf &amp; vbLf &amp; c.Insert &amp; vbLf _
                     &amp; c.Remove &amp; vbLf &amp; c.Substring &amp; vbLf _
                     &amp; c.ToUpper &amp; vbLf &amp; c.ToLower &amp; vbLf _
                     &amp; c.TrimStart &amp; " " &amp; c.TrimEnd)
  Next c
 End Using

' </snippetlinqexamples51>

File: ce/customerengagement/on-premises/snippets/csharp/CRMV8/query/cs/linqexamples51.cs
// <snippetlinqexamples51>


 using (ServiceContext svcContext = new ServiceContext(_serviceProxy))
 {
  var query_string = from c in svcContext.ContactSet
                     where c.ContactId == _contactId2
                     select new
                     {
                      IndexOf = c.FirstName.IndexOf("contact"),
                      Insert = c.FirstName.Insert(1, "Insert"),
                      Remove = c.FirstName.Remove(1, 1),
                      Substring = c.FirstName.Substring(1, 1),
                      ToUpper = c.FirstName.ToUpper(),
                      ToLower = c.FirstName.ToLower(),
                      TrimStart = c.FirstName.TrimStart(),
                      TrimEnd = c.FirstName.TrimEnd(),
                     };

  foreach (var c in query_string)
  {
   System.Console.WriteLine(c.IndexOf + "\n" + c.Insert + "\n" + 
    c.Remove + "\n" + c.Substring + "\n"
                            + c.ToUpper + "\n" + c.ToLower + 
                            "\n" + c.TrimStart + " " + c.TrimEnd);
  }
 }

// </snippetlinqexamples51>